relentless/optimize/objective.py

```diff
@@ -165,3 +168,55 @@ def design_variables(self):
         :class:`ObjectiveFunction` for which the result was constructed, mapped
         to the value of the variables at the time the result was constructed."""
         return self._design_variables
+
+class RelativeEntropy(ObjectiveFunction):
+    def __init__(self, target, simulation, potentials, thermo, dr, communicator=None):
+        self.target = target
+        self.simulation = simulation
+        self.potentials = potentials
+        self.thermo = thermo
+        self.dr = dr
+        self.communicator = communicator
+
+    def compute(self, directory=None):
+        sim = self.simulation.run(self.target, self.potentials, directory, self.communicator)
+        sim_ens = self.thermo.extract_ensemble(sim)
+
+        # compute the relative entropy gradient by integration
+        g_tgt = self.target.rdf
+        g_sim = sim_ens.rdf
+        gradient = {}
+        dvars = self.design_variables()
+
+        for x in dvars:
+            update = 0
+            for i,j in self.potentials.pair.potentials.coeff.pairs:
```

Comment:
We need a simpler way to do this... this could work:
```suggestion
            for i,j in self.target.pairs:
```